analyzer/runtime/evm/erc721.go

```diff
@@ -108,20 +107,46 @@ func evmDownloadNFTERC721Metadata(ctx context.Context, logger *log.Logger, sourc
 		)
 		return nil
 	}
+	// 1. Read metadata into interface{} to allow any syntactically correct
+	//    JSON value.
+	var metadataAny interface{}
 	limitedReader := io.LimitReader(rc, MaxMetadataBytes)
-	var metadataBuilder strings.Builder
-	teeReader := io.TeeReader(limitedReader, &metadataBuilder)
-	var metadata ERC721AssetMetadata
-	if err = json.NewDecoder(teeReader).Decode(&metadata); err != nil {
-		logger.Info("error decoding token metadata",
+	if err = json.NewDecoder(limitedReader).Decode(&metadataAny); err != nil {
+		logger.Info("error decoding token metadata as any",
 			"uri", nftData.MetadataURI,
 			"err", err,
 		)
+		if err = rc.Close(); err != nil {
+			return fmt.Errorf("closing metadata reader: %w", err)
+		}
+		return nil
```

Comment:
Will this result in one of those deterministic errors, meaning we won't retry? Or will we keep parsing the same JSON and failing every 2 minutes or whatever the analyzer's interval is?

Reply:
this is outside of the "collect deterministic errors" zone, it's just logged for our information and entirely treated as success. "we successfully got the metadata and extracted what we could from it. which was nothing because it was unparseable, but still"